### PR TITLE
Update Keychron Q0

### PR DESCRIPTION
**Pull request by adophoxia**
## Description

Redo of https://github.com/qmk/qmk_firmware/pull/21994.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q0/config.h [hunk 1]
```diff
@@ -21,9 +21,10 @@
 /* turn off effects when suspended */
 #define RGB_DISABLE_WHEN_USB_SUSPENDED
 
-/* EEPROM Driver Configuration */
-#define WEAR_LEVELING_LOGICAL_SIZE 2048
-#define WEAR_LEVELING_BACKING_SIZE (WEAR_LEVELING_LOGICAL_SIZE * 2)
+/* Encoder Configuration*/
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (fauxpark) [inline keyboards/keychron/q0/config.h:27]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3
```
Shouldn't do anything if encoder is disabled.

File: keyboards/keychron/q0/rev_0131/config.h [hunk 2]
```diff
@@ -21,13 +21,10 @@
 #define DRIVER_ADDR_1 0b1110111
 
 /* RGB Matrix Configuration */
-#define DRIVER_1_LED_TOTAL 26
-#define RGB_MATRIX_LED_COUNT DRIVER_1_LED_TOTAL
+#define RGB_MATRIX_LED_COUNT 26
 
+/* Scan phase of led driver set as MSKPHASE_9CHANNEL(defined as 0x03 in CKLED2001.h) */
```
Comment (fauxpark) [inline keyboards/keychron/q0/rev_0131/config.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/config.h (hunk 1 above)
Comment (adophoxia) [reply keyboards/keychron/q0/config.h:27]: Would it be better if I'd just move `#define ENCODER_DEFAULT_POS 0x3` into `q0/rev_0131/config.h` instead?

File: keyboards/keychron/q0/rev_0131/readme.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
-# The enhanced version with encoder of the Keychron Q0
+# Keychron Q0 Plus
 
-- The name of this keyboard is "Q0 Plus"
-- Enable EC11 rotary encoder
-- Turn clockwise to increase volume and turn anti-clockwise to decrease volume
-- Press the knob button to mute
+![Keychron Q0 Plus](https://i.imgur.com/EDbfVVO.jpg[/img])
```
Comment (dunk2k) [inline keyboards/keychron/q0/rev_0131/readme.md:3]:
```suggestion
![Keychron Q0 Plus](https://i.imgur.com/EDbfVVOh.jpg)
```

File: keyboards/keychron/q0/readme.md [hunk 4]
```diff
@@ -1,18 +1,22 @@
 # Keychron Q0
 
+![Keychron Q0](https://i.imgur.com/cLbEiZ0.jpg)
```
Comment (dunk2k) [inline keyboards/keychron/q0/readme.md:3]:
```suggestion
![Keychron Q0](https://i.imgur.com/cLbEiZ0h.jpg)
```

File: keyboards/keychron/q0/rev_0130/info.json
Comment (dunk2k) [inline keyboards/keychron/q0/rev_0130/info.json:1]: could Community Layout support for numpad_6x4 be added?
`community_layouts: ["numpad_6x4"],`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/rev_0130/info.json
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/info.json:1]: It's already a default layout option. Is there a need to replace it with the `community_layout` option?

File: keyboards/keychron/q0/config.h (hunk 1 above)
Comment (drashna) [reply keyboards/keychron/q0/config.h:27]: Shouldn't matter, but if it's needed for multiple versions, would be simpler here.

**Review by drashna — COMMENTED** [review]
the revs here are encoder vs non-encoder, correct? 

If so, it would be ideal to rename the folders

**Comment by adophoxia** [comment]
> the revs here are encoder vs non-encoder, correct?
> 
> If so, it would be ideal to rename the folders

They are. As for the folder renaming, wouldn't rlly know what to do about that since I can't use names that are shared with `LAYOUT` names.

File: keyboards/keychron/q0/config.h (hunk 1 above)
Comment (adophoxia) [reply keyboards/keychron/q0/config.h]: Dealt with https://github.com/qmk/qmk_firmware/pull/22068/commits/6060f897b076497dff3fe64b5023ee02e3c2636d.

**Review by fauxpark — APPROVED** [review]

File: keyboards/keychron/q0/rev_0130/info.json
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/info.json:1]: Taken care of with https://github.com/qmk/qmk_firmware/pull/22068/commits/d11da612823d8c9714ed6195cf585d0f6690e068.

**Review by dunk2k — APPROVED** [review]

**Comment by drashna** [comment]
> > the revs here are encoder vs non-encoder, correct?
> > If so, it would be ideal to rename the folders
> 
> They are. As for the folder renaming, wouldn't rlly know what to do about that since I can't use names that are shared with `LAYOUT` names.

I hate to ask this, but after looking at their pages, it looks like plus and non-plus the difference is an encoder and a macro row.  

for reference: 
https://www.keychron.com/products/keychron-q0-qmk-custom-number-pad?_pos=2&_sid=d4acbf2a6&_ss=r
https://www.keychron.com/products/keychron-q0-plus-qmk-custom-number-pad?_pos=3&_sid=d4acbf2a6&_ss=r

To stay consistent with revision/subfolder naming elsewhere, I think that `base` and `plus` would be better naming here.

**Comment by adophoxia** [comment]
> To stay consistent with revision/subfolder naming elsewhere, I think that `base` and `plus` would be better naming here.

I'll take care of that right now.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
sorry, I missed it, but could you add an entry for the keyboard alias json?

**Comment by adophoxia** [comment]
Just got off PC. I'll add it when I can tomorrow.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**